lib/addons/topics-api.ts

```diff
@@ -0,0 +1,40 @@
+import OptableSDK from "../sdk";
+
+declare module "../sdk" {
+    export interface OptableSDK {
+        tryTopicsAPI: () => void;
+    }
+}
+
+declare global {
+    interface Document {
+        browsingTopics: () => Promise<Array<{
+            configVersion: string;
+            modelVersion: string;
+            taxonomyVersion: string;
+            topic: number;
+            version: string;
+        }>>;
+        featurePolicy:{ allowsFeature: (feature:string) => boolean };
+    }
+}
+
+OptableSDK.prototype.tryTopicsAPI = async function () {
+    if (!sessionStorage.topics_fetched && 'browsingTopics' in document && document.featurePolicy.allowsFeature('browsing-topics')) {
+        const topicsArray = await document.browsingTopics();
+        if (topicsArray.length > 0) {
```

Comment:
This check is a bit unnecessary, if it's empty it will loop over an empty list which is a noop.
If you want to keep the check I'd suggest doing an early return to flatten the code.

`if (!topicsArray.length) { return }`